ee/codegen/src/__test__/utils/typing.test.ts

```diff
@@ -0,0 +1,68 @@
+import { describe, it, expect } from "vitest";
+
+import { assertUnreachable, isDefined, isNilOrEmpty } from "src/utils/typing";
+
+describe("isDefined", () => {
+  it("should return true for defined values", () => {
+    expect(isDefined(0)).toBe(true);
+    expect(isDefined(false)).toBe(true);
+    expect(isDefined("")).toBe(true);
+    expect(isDefined([])).toBe(true);
+    expect(isDefined({})).toBe(true);
+  });
```

Comment:
nit: Using `it.each` and parametrizing this test and the is the preferred pattern (same for the ones below where easy).